persistent-mysql/Database/Persist/MySQL.hs

```diff
@@ -21,68 +19,62 @@ module Database.Persist.MySQL
      -- * @ON DUPLICATE KEY UPDATE@ Functionality
     , insertOnDuplicateKeyUpdate
     , insertManyOnDuplicateKeyUpdate
-#if MIN_VERSION_base(4,7,0)
     , HandleUpdateCollision
     , pattern SomeField
-#elif MIN_VERSION_base(4,9,0)
-    , HandleUpdateCollision(SomeField)
-#endif
     , SomeField
```

Comment:
Not sure what the CPP above should do? Just kept the most stuff in there to be sure.

Reply:
My guess is that view patterns are only supported on GHC 8 or above? In any case we have dropped support for pre-GHC 8 so we should be fine there.

Reply:
I've never seen the `pattern` export before. Not sure what that does. :/

Reply:
It exports a pattern synonym. The `pattern` keyword is necessary to distinguish it from the type and data constructor namespace.